### PR TITLE
Feature/generic previewer

### PR DESCRIPTION
**Pull request by phpfs**
This PR implements a generic experiment and dataset previewer.

### CODE REVIEW

File: app/src/components/DatasetCard/DatasetCard.tsx [hunk 1]
```diff
@@ -70,14 +67,18 @@ const DatasetCard = ({
             <IonChip
               color="dark"
               outline={false}
-              onClick={() => uploadedCount && setDatasetPreviewIsOpen(true)}
+              onClick={(): void =>
+                uploadedCount !== undefined ? previewDataset() : undefined
```
Comment (martingraf4) [inline app/src/components/DatasetCard/DatasetCard.tsx:71]: I would use an if statement

File: app/src/components/DatasetCard/DatasetCard.tsx [hunk 2]
```diff
@@ -70,14 +67,18 @@ const DatasetCard = ({
...
+              }
             >
               <IonLabel>Total: {totalCount ?? 'unknown'}</IonLabel>
             </IonChip>
             <IonChip
               color="dark"
               outline={false}
-              onClick={() => uploadedCount && setDatasetPreviewIsOpen(true)}
+              onClick={(): void =>
+                uploadedCount !== undefined ? previewDataset() : undefined
```
Comment (martingraf4) [inline app/src/components/DatasetCard/DatasetCard.tsx:80]: I would use an if statement

File: app/src/components/DatasetCard/DatasetCard.tsx [hunk 3]
```diff
@@ -70,14 +67,18 @@ const DatasetCard = ({
...
+              }
             >
               <IonLabel>Total: {totalCount ?? 'unknown'}</IonLabel>
             </IonChip>
             <IonChip
```
Comment (martingraf4) [inline app/src/components/DatasetCard/DatasetCard.tsx:76]: maybe we should disable if not clickable (also all others)

File: app/src/components/ExperimentCard/ExperimentCard.tsx [hunk 4]
```diff
@@ -43,12 +41,14 @@ const ExperimentCard = ({
             <IonChip
               color="dark"
               outline={false}
-              onClick={() => setShowExperimentPreview(true)}
+              onClick={(): void =>
+                numberOfRecords !== undefined ? previewExperiment() : undefined
```
Comment (martingraf4) [inline app/src/components/ExperimentCard/ExperimentCard.tsx:45]: if statement

**Review by florian-papsdorf — APPROVED** [review]
Nice improvement and it turns more to our existing architecture

File: app/src/store/actions/ExperimentPreviewerActions.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+): easyPrimitiveActionReturn =>
+  easyPrimitiveAction({
+    type: ExperimentPreviewerStoreActionTypes.OPEN_DIALOG,
+    payload: anExperiment,
+  });
+
+export const closePreviewer = (): easyPrimitiveActionReturn =>
+  easyPrimitiveAction({
+    type: ExperimentPreviewerStoreActionTypes.CLOSE_DIALOG,
+    // ignores payload
```
Comment (martingraf4) [inline app/src/store/actions/ExperimentPreviewerActions.ts:19]: why do we have this comment everywhere?

**Review by martingraf4 — CHANGES_REQUESTED** [review]

**Pull request merged**